### ISSUE TITLE
[OoB 5.12] sd-driver repository

### ISSUE THREAD

**Issue by maciejbocianski** [ARMmbed/mbed-os-example-sd-driver#9]
Since `sd-driver` is now part of mbed https://github.com/ARMmbed/mbed-os/tree/master/components/storage/blockdevice/COMPONENT_SD step `Add the sd-driver repository: mbed add sd-driver` could be removed. Adding this lib cause config collision(log below)

```
$ mbed compile -m k64f -t GCC_ARM -v
Building project sd-driver (K64F, GCC_ARM)
Scan: sd-driver
Traceback (most recent call last):
  File "C:\Users\macboc01.ARM\work\mbed\dev\examples\sd-driver\mbed-os\tools\make.py", line 74, in wrapped_build_project
    src_dir, build_dir, mcu, *args, **kwargs
  File "C:\Users\macboc01.ARM\work\mbed\dev\examples\sd-driver\mbed-os\tools\build_api.py", line 476, in build_project
    src_paths, toolchain, inc_dirs=inc_dirs)
  File "C:\Users\macboc01.ARM\work\mbed\dev\examples\sd-driver\mbed-os\tools\resources\__init__.py", line 570, in scan_with_toolchain
    toolchain.config.load_resources(self)
  File "C:\Users\macboc01.ARM\work\mbed\dev\examples\sd-driver\mbed-os\tools\config\__init__.py", line 1338, in load_resources
    f.path for f in resources.get_file_refs(FileType.JSON)
  File "C:\Users\macboc01.ARM\work\mbed\dev\examples\sd-driver\mbed-os\tools\config\__init__.py", line 587, in add_config_files
    self.lib_config_data[cfg["name"]]["__config_path"]))
ConfigException: Library name 'sd' is not unique (defined in 'C:\Users\macboc01.ARM\work\mbed\dev\examples\sd-driver\sd-driver\config\mbed_lib.json' and 'C:\Users\macboc01.ARM\work\mbed\dev\examples\sd-driver\mbed-os\components\storage\blockdevice\COMPONENT_SD\mbed_lib.json')
[mbed] Working path "C:\Users\macboc01.ARM\work\mbed\dev\examples\sd-driver" (program)
[mbed] Exec "c:\python27\python.exe -u C:\Users\macboc01.ARM\work\mbed\dev\examples\sd-driver\mbed-os\tools\make.py -t GCC_ARM -m k64f --source . --build .\BUILD\K64F\GCC_ARM -v" in "C:\Users\macboc01.ARM\work\mbed\dev\examples\sd-driver"
[mbed] ERROR: "c:\python27\python.exe" returned error.
       Code: 1
       Path: "C:\Users\macboc01.ARM\work\mbed\dev\examples\sd-driver"
       Command: "c:\python27\python.exe -u C:\Users\macboc01.ARM\work\mbed\dev\examples\sd-driver\mbed-os\tools\make.py -t GCC_ARM -m k64f --source . --build .\BUILD\K64F\GCC_ARM -v"
       Tip: You could retry the last command with "-v" flag for verbose output
```

**Comment by LDong-Arm** [comment]
This issue still exists as of OoB 6.0

**Comment by adbridge** [comment]
@evedon @bulislaw who can we get to look at this and is it a priority for 6.0?

**Comment by evedon** [comment]
It is a documentation issue.

**Comment by LDong-Arm** [comment]
@evedon https://github.com/ARMmbed/mbed-os-example-sd-driver/pull/35 created

**Comment by ciarmcom** [comment]
@maciejbocianski thank you for raising this issue.Please take a look at the following comments:

Could you add some more detail to the description? A good description should be at least 25 words.
What target(s) are you using?
What toolchain(s) are you using?
What version of Mbed OS are you using (tag or sha)?
It would help if you could also specify the versions of any tools you are using?
How can we reproduce your issue?

NOTE: If there are fields which are not applicable then please just add 'n/a' or 'None'.This indicates to us that at least all the fields have been considered.
Please update the issue header with the missing information, the issue will not be mirroredto our internal defect tracking system or investigated until this has been fully resolved.

**Comment by ciarmcom** [comment]
@maciejbocianski it has been 5 days since the last reminder. Could you please update the issue header as previously requested?

**Comment by ciarmcom** [comment]
@maciejbocianski it has been 5 days since the last reminder. Could you please update the issue header as previously requested?

**Comment by ciarmcom** [comment]
@maciejbocianski it has been 5 days since the last reminder. Could you please update the issue header as previously requested?

**Comment by ciarmcom** [comment]
@maciejbocianski it has been 5 days since the last reminder. Could you please update the issue header as previously requested?

**Comment by ciarmcom** [comment]
@maciejbocianski it has been 5 days since the last reminder. Could you please update the issue header as previously requested?

**Comment by ciarmcom** [comment]
@maciejbocianski it has been 5 days since the last reminder. Could you please update the issue header as previously requested?

**Comment by ciarmcom** [comment]
@maciejbocianski it has been 5 days since the last reminder. Could you please update the issue header as previously requested?

**Comment by evedon** [comment]
Issue fixed in #35
